detect-nuget-inspector/detect-nuget-inspector/Inspection/Util/SolutionDirectoryBuildPropertyLoader.cs

```diff
@@ -70,6 +70,23 @@ private HashSet<PackageId> GetPackageReference(XmlNodeList packageNodes)
                     {
                         packageReferences.Add(new PackageId(name, version));
                     }
+
+                    if (String.IsNullOrWhiteSpace(version))
+                    {
+                        foreach (XmlNode node in packageNode.ChildNodes)
+                        {
+                            if (node.Name == "Version")
+                            {
+                                version = node.InnerXml;
+                                break;
+                            }
+                        }
+                        
+                        if (!String.IsNullOrWhiteSpace(name) && !String.IsNullOrWhiteSpace(version))
+                        {
+                            packageReferences.Add(new PackageId(name, version));
+                        }
+                    }
```

Comment:
similar comment to above except with this block on 85 and the one on 69

Reply:
Related: the changes are also very similar across the two files. Is it worthwhile to try to consolidate those as well?

Reply:
I am afraid we cannot do that at this point, the classes and the results for both the files you are looking at are referenced at different places. Trying to consolidate them at this point would lead to a major rewrite of the whole code which we are already planning to do at later point of time.